apisix/init.lua

```diff
@@ -266,12 +266,21 @@ local function set_upstream_host(api_ctx, picked_server)
 end
 
 
+local upstream_proxy_headers = {
+    var_x_forwarded_proto = 'X-Forwarded-Proto',
+    var_x_forwarded_for = 'X-Forwarded-For',
+    var_x_forwarded_host = 'X-Forwarded-Host',
+    var_x_forwarded_port = 'X-Forwarded-Port',
+}
+
 local function set_upstream_headers(api_ctx, picked_server)
     set_upstream_host(api_ctx, picked_server)
 
-    local hdr = core.request.header(api_ctx, "X-Forwarded-Proto")
-    if hdr then
-        api_ctx.var.var_x_forwarded_proto = hdr
+    for ngx_key, hdr_key in pairs(upstream_proxy_headers) do
```

Comment:
We can also remove https://github.com/apache/apisix/blob/a0efeca194bcca58e62d73c0a19464f480a1b7c9/apisix/cli/ngx_tpl.lua#L600 and https://github.com/apache/apisix/blob/a0efeca194bcca58e62d73c0a19464f480a1b7c9/t/APISIX.pm#L697

Reply:
Can you please explain why the requested changes are needed?

Reply:
The same feature is implemented in your PR, except for the  X-Forwarded-For one. After your change, there won't be client IP appending to the http_x_forwarded_for  when the header is not rewritten. We should add test for it and fix it.